cartography/intel/keycloak/authenticationexecutions.py

```diff
@@ -0,0 +1,322 @@
+    transformed_by_id: OrderedDict[str, dict[str, Any]] = OrderedDict()
+    initial_flow_steps: list[dict[str, str]] = []
+    flow_steps: list[dict[str, str]] = []
+
+    for flow_name, executions in exec_by_flow.items():
+        _parent_by_level: dict[int, str] = {}
+        _root_executions: list[dict[str, Any]] = []
+
+        # Transform executions to include parent flow/subflow relationships
+        # and create a hierarchical structure for graph processing
+        for execution in executions:
+            # Level 0 executions belong directly to the named flow
+            if execution["level"] == 0:
+                execution["_parent_flow"] = flow_name
+                _root_executions.append(execution)
+            else:
+                # Nested executions belong to their parent subflow
+                execution["_parent_subflow"] = _parent_by_level[execution["level"] - 1]
+                transformed_by_id[execution["_parent_subflow"]]["_children"].append(
+                    execution
+                )
+
+            # Track subflow parents for the next nesting level
+            if execution.get("authenticationFlow", True):
+                _parent_by_level[execution["level"]] = execution["id"]
+
+            execution["_children"] = []
+            execution["is_terminal_step"] = False  # Placeholder for terminal step flag
+            transformed_by_id[execution["id"]] = execution
+
+        # Process authentication flow structure and build execution graph
+        # Reference: https://www.keycloak.org/docs/latest/server_admin/index.html#_execution-requirements
+        entries, links, terminals = _recursive_transform_flow(_root_executions)
+
+        for entry in entries:
+            initial_flow_steps.append(
+                {
+                    "flow_name": flow_name,
+                    "execution_id": entry,
+                    "realm": realm,
+                }
+            )
+
+        for link in links:
+            flow_steps.append(
+                {
+                    "source": link[0],
+                    "target": link[1],
+                }
+            )
+
+        for node_id in terminals:
+            transformed_by_id[node_id]["is_terminal_step"] = True
+
+    return list(transformed_by_id.values()), flow_steps, initial_flow_steps
+
+
+@timeit
+def load_authenticationexecutions(
+    neo4j_session: neo4j.Session,
+    data: list[dict[str, Any]],
+    realm: str,
+    update_tag: int,
+) -> None:
+    logger.info(
+        "Loading %d Keycloak AuthenticationExecutions (%s) into Neo4j.",
+        len(data),
+        realm,
+    )
+    load(
+        neo4j_session,
+        KeycloakAuthenticationExecutionSchema(),
+        data,
+        LASTUPDATED=update_tag,
+        REALM=realm,
+    )
+
+
+def load_execution_flow(
+    neo4j_session: neo4j.Session,
+    flow_steps: list[dict[str, Any]],
+    initial_flow_steps: list[dict[str, str]],
+    realm_id: str,
+    update_tag: int,
+) -> None:
```

Comment:
[nonblock] a comment here on why we're doing matchlink instead of having it as part of the standard data model would be helpful. I'm assuming this is a bit of an analysis step